src/store/elementalChat.js

```diff
@@ -384,6 +384,7 @@ export default {
       }
 
       const activeChatters = uniqBy((channel.messages || []).map(message => message.createdBy), arrayBufferToBase64)
+        .filter(userIdBuffer => arrayBufferToBase64(userIdBuffer) !== arrayBufferToBase64(agentKey))
```

Comment:
This feels a little inefficient to me so I'm gonna try micro-optimizing it
```suggestion
      const activeChattersMap = {}
      for (const message of channel.messages || []) {
          const userIdBuffer = message.createdBy
          const base64 = arrayBufferToBase64(userIdBuffer)
          activeChattersMap[base64] = userIdBuffer
      }
      
      // Don't send signals to ourselves
      const agentKeyBase64 = arrayBufferToBase64(agentKey)
      if (activeChattersMap[agentKeyBase64] !== undefined) {
          delete activeChattersMap[agentKeyBase64]
      }
      
      const activeChatters = Object.values(activeChattersMap)
```

Reply:
That's def more efficient (n vs 3n). 

The other things I'd consider are readability, maintainability and likelihood of bugs.

For me, the functional style generally wins hands down in those categories, including here, so I would want to feel a compelling reason to optimize.

This code is called every time the current channel is update, so basically every time you send or receive a message. n is going to be on the order of 50. 50 loops vs 150 loops once every few seconds, I'm not sure where that falls. I don't think it's unreasonable to want to optimize there.

Ideally we're profiling the code so that we can just see where we should be optimizing.

Reply:
yeah... I agree about the tradeoff. That's part of why I really like Rust. Often I can write code in the readable style without compromising on efficiency. I'll leave it up to you